specs/latest/1.0/index.html

```diff
@@ -3462,12 +3462,22 @@ <h3><a name="BUFFER_OBJECT_BINDING">Buffer Object Binding</a></h3>
     <h3>No Client Side Arrays</h3>
 
 <p>
+The WebGL API does not support client-side arrays.
 
-The WebGL API does not support client-side arrays. If <code>vertexAttribPointer</code> is called
-without a <code>WebGLBuffer</code> bound to the <code>ARRAY_BUFFER</code> binding point, and
-<code>offset</code> is non-zero, an <code>INVALID_OPERATION</code> error is generated. If
-<code>drawElements</code> is called with a <code>count</code> greater than zero, and no
-<code>WebGLBuffer</code> is bound to the <code>ELEMENT_ARRAY_BUFFER</code> binding point, an
+<p>
+If a vertex attribute is enabled as an array via <code>enableVertexAttribArray</code> but no buffer
+is bound to that attribute (generally via <code>bindBuffer</code> and
+<code>vertexAttribPointer</code>), then draw commands (<code>drawArrays</code> or
+<code>drawElements</code>) will generate an <code>INVALID_OPERATION</code> error.
+
+<p>
+If an indexed draw command (<code>drawElements</code>) is called and no <code>WebGLBuffer</code> is
+bound to the <code>ELEMENT_ARRAY_BUFFER</code> binding point, an <code>INVALID_OPERATION</code>
+error is generated.
```

Comment:
Per above – I'm guessing it's not possible to bypass this one check only for zero-length draw calls?

Reply:
We could, but it breaks pattern with how we handle null-buffer vertexAttribArrays. IMO it's cleaner to just always do it?